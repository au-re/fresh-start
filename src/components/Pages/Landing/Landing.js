import React, { Component } from "react";
import styled from "styled-components";

import { fadeColor, fadeIn } from "../../../styles/animations";

const LandingTitle = styled.h1`
  color: white;
  align-self: center;
  text-transform: uppercase;
  opacity: 0;
  animation: ${fadeIn()} 0.5s ease-out forwards;
  animation-delay: 0.5s;
`;

const LoadingLanding = styled.div`
  background: salmon;
`;

/* the landing page */

class Landing extends Component {

  static Title = LandingTitle;
  static LoadingLanding = LoadingLanding;

  render() {
    const { className } = this.props;

    return (
      <div className={className}>
        <LandingTitle>I'm just a Boilerplate</LandingTitle>
      </div>);
  }
}

export default styled(Landing) `
  background: salmon;
  color: white;
  height: 100%;
  display: flex;
  justify-content: center;
  animation: ${fadeColor("salmon", "tomato")} 1s ease-out forwards;
`;
