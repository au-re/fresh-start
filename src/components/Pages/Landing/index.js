import React, { Component } from "react";

import Button from "../../Button";

class Landing extends Component {
  render() {
    const { className } = this.props;
    return (
      <div className={className}>
        landing page
        <Button>hello world</Button>
      </div>);
  }
}

export default Landing;
