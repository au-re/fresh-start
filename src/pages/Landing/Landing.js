import get from "lodash.get";
import React, { Component } from "react";
import Helmet from "react-helmet";
import styled from "styled-components";

import GithubCorner from "../../components/GithubCorner/GithubCorner";
import ThemeMenu from "../../components/ThemeMenu/ThemeMenu";
import config from "../../config";
import { fadeBackgroundColor, fadeIn } from "../../styles";

const LandingTitle = styled.h1`
  color: ${({ theme }) => get(theme, "blends.landing.text")};
  font-weight: bold;
  font-size: 1.5rem;
  align-self: center;
  text-align: center;
  text-transform: uppercase;
  opacity: 0;
  animation: ${fadeIn()} 0.5s ease-out forwards;
  animation-delay: 0.5s;
`;

const LandingMenu = styled.div`
  align-self: center;
`;

const LandingFooter = styled.div`
  position: absolute;
  padding: 1rem;
  bottom: 0;
  left: 0;
`;

/* the landing page */

class Landing extends Component {

  render() {
    const { className } = this.props;

    return (
      <div className={className}>
        <Helmet>
          <title>Landing</title>
        </Helmet>
        <GithubCorner />
        <LandingTitle>I'm just a Boilerplate</LandingTitle>
        <LandingMenu>
          themeable
          <ThemeMenu />
        </LandingMenu>
        <LandingFooter>{config.copyrightNotice}<strong>{config.company}</strong></LandingFooter>
      </div>);
  }
}

export default styled(Landing)`
  background: ${({ theme }) => get(theme, "blends.landing.background")};
  color: ${({ theme }) => get(theme, "blends.landing.color")};
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  animation: ${({ theme }) => fadeBackgroundColor(
    get(theme, "blends.landing.backgroundIntro"),
    get(theme, "blends.landing.background"))} 1s ease-out forwards;
`;
