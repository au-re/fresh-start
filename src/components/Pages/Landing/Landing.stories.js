import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import Landing from "./Landing";

storiesOf("Landing", module)

  .add("basic usage", withInfo(`A landing page`)(
    () => <Landing />));
