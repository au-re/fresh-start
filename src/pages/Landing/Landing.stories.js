import { storiesOf } from "@storybook/react";
import React from "react";
import { MemoryRouter as Router } from "react-router";

import ThemeProvider from "../../components/ThemeProvider/ThemeProvider";
import themes from "../../themes";
import Landing from "./Landing";

storiesOf("Landing Page", module)
  .add("basic usage", () =>
    <Router>
      <ThemeProvider themes={themes}>
        <Landing />
      </ThemeProvider>
    </Router>);
