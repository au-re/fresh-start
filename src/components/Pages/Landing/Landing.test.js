import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

import Landing from "./Landing";

Enzyme.configure({ adapter: new Adapter() });

describe("Landing", () => {

  it("renders without props", () => {
    expect(<Landing />).toMatchSnapshot();
  });
});
