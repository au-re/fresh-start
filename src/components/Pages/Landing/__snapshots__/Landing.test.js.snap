// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders without props 1`] = `
.c1 {
  color: white;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  text-transform: uppercase;
  opacity: 0;
  -webkit-animation: eMLfYp 0.5s ease-out forwards;
  animation: eMLfYp 0.5s ease-out forwards;
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
}

.c0 {
  background: salmon;
  color: white;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-animation: dZYEHQ 1s ease-out forwards;
  animation: dZYEHQ 1s ease-out forwards;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    I'm just a Boilerplate
  </h1>
</div>
`;
